boa/contracts/vyper/vyper_contract.py

```diff
@@ -1048,7 +1048,23 @@ def __init__(self, contract, fn_source):
         self._source_map = source_map
 
 
-_typ_cache = {}
+_str_wrapper = type("str_wrapper", (str,), {})
+
+
+class _AddressWrapper(_str_wrapper):
+    """
+    A wrapper around str that adds a canonical_address property to mimic `Address`.
+    This is necessary because `Address` overrides `__new__`, so it always returns
+    an Address instance that cannot be modified.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        self = super().__new__(cls, *args, **kwargs)
+        self.canonical_address = Address(self).canonical_address
+        return self
```

Comment:
yea after thinking about this i don't think it's necessary. see the special handling for bool in `vyper_object()`, we can follow that for the `Address` type.